nncf/openvino/engine.py

```diff
@@ -29,28 +29,6 @@ class OVCompiledModelEngine(Engine):
 
     def __init__(self, model: ov.CompiledModel):
         self.compiled_model = model
-        self.input_tensor_names = set()
-        self.number_of_inputs = len(model.inputs)
-        for model_input in model.inputs:
-            self.input_tensor_names.update(model_input.get_names())
-
-    def _check_input_data_format(
-        self, input_data: Union[np.ndarray, List[np.ndarray], Tuple[np.ndarray], Dict[str, np.ndarray]]
-    ) -> None:
-        """
```

Comment:
There are two reasons behind:
1) Based on our samples for OV we provide `torch.Tensor` as an input for OVEngine (which works fine), which is not specified in typehints and, moreover, should be covered as well as `np.ndarray`. I mean this check should be extended with a `torch.Tensor`, or in case batch_size > 1 it will not work correctly.
`actual_num_inputs = 1 if isinstance(input_data, np.ndarray) else len(input_data)`. 
In my undertanding, to do that there is a need to add `torch` dependency for OpenVINO backend, which I think we do not want to do.
2) Personally, I don't see a reason of checking input for OVEngine, as if the input is incorrect the OpenVINO raises an error by itself with a corresponding comment.

Reply:
CC: @KodiaqQ

Reply:
Yes, there is an error message from OpenVINO in that case. 
Tested on mobilenet_v2 sample:
For extra name "incorrect_input_name" in input tensor such error occurs:
```
Check '::getPort(port, name, {_impl->get_inputs(), _impl->get_outputs()})' failed at src/inference/src/infer_request.cpp:198:
Port for tensor name incorrect_input_name was not found.
```
For empty tensor such error occurs:
```
RuntimeError: Exception from src/inference/src/infer_request.cpp:231:
[ GENERAL_ERROR ] Shape inference of Convolution node with name /features/features.0/features.0.0/Conv/WithoutBiases failed: Check 'data_rank.is_dynamic() || filters_rank.is_dynamic() || data_shape[1].compatible(filters_shape[1])' failed at src/core/shape_inference/include/convolution_shape_inference_util.hpp:309:
While validating node 'opset1::Convolution /features/features.0/features.0.0/Conv/WithoutBiases (input.1[0]:f32[?,3,224,224], onnx::Conv_538[0]:f32[32,3,3,3]) -> (f32[?,32,112,112])' with friendly_name '/features/features.0/features.0.0/Conv/WithoutBiases':
Data batch channel count (0) does not match filter input channel count (3).

```
For me looks okay. Suggest to remove this check in NNCF

Reply:
I think we should rely on the third opinion there. I do not support messages and checks removing.
@alexsu52, what do you think?

Reply:
Please, note, that keeping this logic results at adding `isinstance` check for `torch.Tensor` as well as for `numpy`.

Reply:
Since when OpenVINO support `torch.Tensor` as the input for the model?

Reply:
Have no idea, but look at our examples. `torch.Tensor` is provided there as an input. 
https://github.com/openvinotoolkit/nncf/blob/develop/examples/post_training_quantization/openvino/mobilenet_v2/main.py#L103

Reply:
That is weird. In this case, we should also update `nncf/openvino/engine.py` with `torch.Tensor`. And then check the TF format.

Reply:
I don't think that we should check TF format. But for `torch.Tensor` the current logic should be updated if we want to keep `_check_input_data_format`. 
My suggestion is to remove it.